src/omv/imlib/framebuffer.c

```diff
@@ -227,19 +239,56 @@ uint32_t framebuffer_get_frame_size()
     return image_size(&img);
 }
 
+// Returns the number of bytes the frame buffer could be at the current moment it time.
+static int framebuffer_raw_buffer_size()
+{
+    int size = (int) (fb_alloc_stack_pointer() - ((char *) framebuffer->data));
+    // We don't want to give all of the frame buffer RAM to the frame buffer. So, we will limit
+    // the maximum amount of RAM we return.
+    return IM_MIN(size, OMV_RAW_BUF_SIZE);
+}
+
 uint32_t framebuffer_get_buffer_size()
 {
-    uint32_t size = (uint32_t) (fb_alloc_stack_pointer() - ((char *) framebuffer->pixels));
-    // We don't want to give all of the frame buffer RAM to the frame buffer. So, we will limit the
-    // maximum amount of RAM we return.
-    size = IM_MIN(size, OMV_RAW_BUF_SIZE);
-    // Needs to be a multiple of 32 for DMA transfers...
-    return (size / 32) * 32;
+    int size;
+
+    if (vbuffer_count == 1) {
+        // With only 1 vbuffer it's fine to allow the frame buffer size to change given fb_alloc().
+        size = framebuffer_raw_buffer_size();
+    } else {
+        // Whatever the raw size was when the number of buffers were set is locked in...
+        size = vbuffer_raw_size;
+    }
+
+    // Remove the size of the state header plus alignment padding.
+    size -= VBUFFER_STATE_SIZE;
```

Comment:
```suggestion
    // Remove the size of the vbuffer header.
    size = vbuffer_raw_size - (vbuffer_count == 1) ? 0 : VBUFFER_STATE_SIZE;
```
`vbuffer_raw_size` is already set to `framebuffer_raw_buffer_size() / 1` if `vbuffer_count == 1`

Reply:
This suggestion breaks the code. For the 1 buffer case fb_alloc is allowed to creep into the frame buffer. It's limited how far it can go by framebuffer_total_buffer_size(). Similarly, framebuffer_raw_buffer_size() limits the framebuffer from growing into fb_alloc. However, in the multibuffer case we can't do this anymore so we have to lock the size when the number of buffers are set with vbuffer_raw_size.